lib/streamlit/runtime/pages_manager.py

```diff
@@ -276,6 +276,7 @@ def get_pages(self) -> dict[PageHash, PageInfo]:
 
             pages = self.pages_strategy.get_pages()
             self._cached_pages = pages
+            self._on_pages_changed.send()
```

Comment:
I don't think this is the right place to call this given that moving the call here doesn't remove it from being called in V2, it just moves it to happen in a place where the weird behavior due to it is less noticeable.

I'm wondering if we really need this call at all. There's also a call to `self._on_pages_changed.send()` in `invalidate_pages_cache`, so I don't think we need another one here.

Also, I think the whole notion of page cache invalidation only really makes sense in V1, so the natural place to move all of the pages changed callback related things would be within the V1 strategy class

Reply:
I agree. I was thinking of following up with this change, but I can address this here.